pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -179,6 +180,11 @@ func (gc *GarbageCollector) Sync(discoveryClient discovery.ServerResourcesInterf
 		// Get the current resource list from discovery.
 		newResources := GetDeletableResources(discoveryClient)
 
+		// Filter resources that have been uninstalled and reinstalled on the cluster since the last sync.
+		// The following sync will recognize them as new resources and they will be restarted.
+		// If we do not filter them out, the informer that was previously stopped will not get restarted.
+		gc.dependencyGraphBuilder.filterStoppedResources(newResources)
```

Comment:
Do I understand this correctly: The uninstalled CRDs are filtered out, so even if they have been reinstalled before this sync, they won't be restarted in this sync, and they need to wait for the next sync.

Reply:
yes that is correct

Reply:
@caesarxuchao could you help review the bug fix in GC? in case i miss anything there.